src/main/java/jenkins/plugins/logstash/persistence/BuildData.java

```diff
@@ -152,6 +152,11 @@ public BuildData(AbstractBuild<?, ?> build, Date currentTime) {
         }
       }
     }
+    try {
+      buildVariables.putAll(build.getEnvironment(listener));
+    } catch (Exception e) {
```

Comment:
good point, can you change it like this: 3f07de719bad0c3008e4385c0e8a78be7bfc18c6 ?